components/draw/GeometryMode.svelte

```diff
@@ -0,0 +1,169 @@
+  export let mode;
+  export let routeSnapper;
+  export let snapTool;
+  export let drawControls;
+
+  export function start() {}
+  export function stop() {
+    if (currentlyEditing) {
+      // We could've been editing anything; just handle all possibilities
+      drawControls.changeMode("static");
+      drawControls.deleteAll();
+
+      routeSnapper.stop();
+
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        delete feature.properties.hide_while_editing;
+        return gj;
+      });
+    }
+
+    currentlyEditing = null;
+    currentHover.set(null);
+  }
+
+  // This is different than the stores.js one!
+  let currentlyEditing = null;
+
+  // Calculate hover
+  $map.on("mousemove", (e) => {
+    if (mode == thisMode && currentlyEditing == null) {
+      let results = $map.queryRenderedFeatures(e.point, {
+        layers: [
+          "interventions-points",
+          "interventions-lines",
+          "interventions-polygons",
+        ],
+      });
+      // TODO ? syntax
+      var newHoverId = null;
+      if (results.length > 0) {
+        newHoverId = results[0].id;
+      }
+      currentHover.set(newHoverId);
+    }
+  });
+  $map.on("mouseout", () => {
+    if (mode == thisMode && currentlyEditing == null) {
+      currentHover.set(null);
+    }
+  });
+
+  // Handle clicking the hovered feature
+  $map.on("click", (e) => {
+    if (mode == thisMode && currentlyEditing == null) {
+      let results = $map.queryRenderedFeatures(e.point, {
+        layers: [
+          "interventions-points",
+          "interventions-lines",
+          "interventions-polygons",
+        ],
+      });
+      if (results.length > 0) {
+        startEditing(results[0].id);
+      }
+    }
+  });
+
+  snapTool.addEventListener("new-route", (e) => {
+    if (mode == thisMode) {
+      const editedRoute = e.detail;
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        // Keep the ID and any properties. Just copy over stuff from routeSnapper.
+        // TODO We're depending on implementation details here and knowing what to copy...
+        feature.properties.length_meters = editedRoute.properties.length_meters;
+        feature.properties.waypoints = editedRoute.properties.waypoints;
+        delete feature.properties.hide_while_editing;
+        feature.geometry = editedRoute.geometry;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+  snapTool.addEventListener("no-new-route", () => {
+    if (mode == thisMode) {
+      // Don't modify the thing we were just editing
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        delete feature.properties.hide_while_editing;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+
+  $map.on("draw.update", (e) => {
```

Comment:
One slight behaviour different I find when using Edit Geometry mode is if I click a Polygon, edit it and try to finish editing by clicking Enter (which is the behaviour for creating a Polygon) we don't drop out of Edit mode, the only way to do that is by clicking away from the editing area (firing next event on line 121)

Reply:
I'd note I don't think this is a blocking issue, just worth noting.

Reply:
Agree this is annoying. I'll aim to address this when we replace polygon drawing.